client/blocks/reader-post-card/compact.jsx

```diff
@@ -1,23 +1,34 @@
 import { useBreakpoint } from '@automattic/viewport-react';
 import clsx from 'clsx';
+import { useTranslate } from 'i18n-calypso';
 import PropTypes from 'prop-types';
 import { useState } from 'react';
 import ReaderExcerpt from 'calypso/blocks/reader-excerpt';
 import ReaderPostEllipsisMenu from 'calypso/blocks/reader-post-options-menu/reader-post-ellipsis-menu';
 import AutoDirection from 'calypso/components/auto-direction';
+import ReaderFollowButton from 'calypso/reader/follow-button';
+import { READER_DISCOVER } from 'calypso/reader/follow-sources';
 import FeaturedAsset from './featured-asset';
 
-const CompactPost = ( {
-	children,
-	post,
-	expandCard,
-	postKey,
-	isExpanded,
-	site,
-	postByline,
-	teams,
-	openSuggestedFollows,
-} ) => {
+const CompactPost = ( props ) => {
+	const {
+		children,
+		post,
+		expandCard,
+		postKey,
+		isExpanded,
+		site,
+		postByline,
+		teams,
+		openSuggestedFollows,
+	} = props;
+
+	// Rather than create complex logic to create context or pass props
+	// to see if the user is on the discover page, let's check the pathname
+	const path = window.location.pathname.split( '/' );
+	const isDiscover = path.length > 0 && path[ 1 ].includes( 'discover' );
```

Comment:
I decided to just check the `window.location` to determine whether we're on the Discover page, and it seems to work well.

The alternative is to somehow provide that information to the `Stream` component's children via the `DiscoverStream` component (below), either through...
1. Creating a new React context
2. Passing an `isDiscover` prop or similar down through the hierarchy (a few layers deep)

Both of those solutions seemed too complex for what they'd be worth. Thoughts?

 https://github.com/Automattic/wp-calypso/blob/01026b73e79245b5f2c2a4c491732849970d9028/client/reader/discover/discover-stream.js#L100-L107

Reply:
I agree that checking the route from `window.location` seems like the best choice here. Can we please move the implementation to an util function `isDiscoverPage`?

Reply:
Done 👍